<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>894</width>
    <height>682</height>
   </rect>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>50</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>CSWeld</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,8">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>6</number>
      </property>
      <item>
       <widget class="QListWidget" name="listWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>154</width>
          <height>0</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">QListWidget {
    min-width: 150px;
    max-width: 150px;
    color: rgb(220, 220, 220); /* 柔和的浅灰白 */
    background: rgb(80,90,100);
    margin:0px,0px,0px,0px;
}

QListWidget::item {
    min-height: 40px;   /*设置item高度*/
    border-style: none;  /*去掉item的borber*/
    color: rgb(255, 255, 255); /*文字颜色*/
}

QListWidget::Item:selected {
    background: lightGray;
    color: rgb(0, 0, 0); /*文字颜色*/
    border-left: 10px solid rgb(12,170,239);/*左侧边缘*/

}</string>
        </property>
        <property name="currentRow">
         <number>-1</number>
        </property>
        <item>
         <property name="text">
          <string>Point Cloud Cropping</string>
         </property>
         <property name="textAlignment">
          <set>AlignCenter</set>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Downsampling</string>
         </property>
         <property name="textAlignment">
          <set>AlignCenter</set>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Measurement</string>
         </property>
         <property name="textAlignment">
          <set>AlignCenter</set>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="data_listWidget">
        <property name="styleSheet">
         <string notr="true">QListWidget {
    min-width: 150px;
    max-width: 150px;
}

QListWidget::item {
    min-height: 20px;   /* 设置item高度 */
    border-style: none; /* 去掉item的border */
    color: rgb(0, 0, 0); /* 文字颜色 */
    qproperty-alignment: AlignCenter; /* 文字居中 */
}

QListWidget::item:selected {
    background: lightGray;
    color: rgb(0, 0, 0); /* 文字颜色 */
    border-left: 15px solid rgb(85, 85, 255); /* 左侧边缘 */
    qproperty-alignment: AlignCenter; /* 确保选中时文字也居中 */
}</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="8,3">
      <item>
       <widget class="QVTKOpenGLWidget" name="qvtkWidget" native="true">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="leftMargin">
          <number>3</number>
         </property>
         <property name="topMargin">
          <number>3</number>
         </property>
         <property name="rightMargin">
          <number>3</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <property name="spacing">
          <number>3</number>
         </property>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0">
        <item>
         <widget class="QTextBrowser" name="LogMessage_textBrowser">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>330</width>
            <height>150</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>352</width>
            <height>155</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>894</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="file_action"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="help_action"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>50</width>
     <height>50</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="home_action"/>
   <addaction name="file_action"/>
   <addaction name="analysis_action"/>
   <addaction name="run_action"/>
   <addaction name="help_action"/>
  </widget>
  <action name="home_action">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/home.png</normaloff>:/image/home.png</iconset>
   </property>
   <property name="text">
    <string>Home</string>
   </property>
   <property name="toolTip">
    <string>主页</string>
   </property>
   <property name="shortcutContext">
    <enum>Qt::WidgetShortcut</enum>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="help_action">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/help.png</normaloff>:/image/help.png</iconset>
   </property>
   <property name="text">
    <string>Help</string>
   </property>
   <property name="toolTip">
    <string>帮助</string>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="analysis_action">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/analysis.png</normaloff>:/image/analysis.png</iconset>
   </property>
   <property name="text">
    <string>Analysis</string>
   </property>
   <property name="toolTip">
    <string>曲面焊缝分析</string>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
  <action name="run_action">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/blue_once.png</normaloff>:/image/blue_once.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>执行选择的算法</string>
   </property>
   <property name="menuRole">
    <enum>QAction::ApplicationSpecificRole</enum>
   </property>
  </action>
  <action name="file_action">
   <property name="icon">
    <iconset resource="res.qrc">
     <normaloff>:/image/open_file.png</normaloff>:/image/open_file.png</iconset>
   </property>
   <property name="text">
    <string>Open File</string>
   </property>
   <property name="toolTip">
    <string>open file</string>
   </property>
   <property name="menuRole">
    <enum>QAction::TextHeuristicRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKOpenGLWidget</class>
   <extends>QWidget</extends>
   <header>qvtkopenglwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res.qrc"/>
 </resources>
 <connections/>
</ui>
